stdlib/public/core/Arrays.swift.gyb

```diff
@@ -1624,23 +1598,22 @@ extension ${Self} {
     return try body(&inoutBufferPointer)
   }
 
-  @discardableResult
   public func _copyContents(
-    initializing ptr: UnsafeMutablePointer<Element>
-  ) -> UnsafeMutablePointer<Element> {
-    if let s = self._baseAddressIfContiguous {
-      let count = self.count
-      ptr.initialize(from: s, count: count)
+    initializing buf: UnsafeMutableBufferPointer<Element>
+  ) -> Iterator? {
+    // FIXME: decide if this really is a fatalError
+    guard var p = buf.baseAddress 
+      else { fatalError("Attempt to copy contents into nil buffer pointer") }
+    if let s = _baseAddressIfContiguous {
+      p.initialize(from: s, count: self.count)
       _fixLifetime(self._owner)
-      return ptr + count
     } else {
-      var p = ptr
       for x in self {
         p.initialize(to: x)
         p += 1
       }
-      return p
     }
+    return nil
```

Comment:
This is confusing. The Sequence._copyContents interface claims to support under-sized buffer arguments. But this implementation clearly doesn't support that. Is there a subtle assumption being made that Array always correctly reports it's count, thus we don't need to handle under-sized buffers? I don't think we can make any assumptions here because UnsafeMutableBufferPointer.initialize<S: Sequence>(from: S) is public.